modules/kms/main.tf

```
@@ -1,15 +1,78 @@
+locals {
+  account_id        = data.aws_caller_identity.current.account_id
+  account_principal = "arn:${data.aws_partition.current.partition}:iam::${local.account_id}:root"
+
+  principals = sort(distinct(concat(
+    var.allowed_principal_arns,
+    module.allowed_role_map.principals,
+  )))
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_partition" "current" {}
+
+module "allowed_role_map" {
+  source = "../account-map/modules/roles-to-principals"
+
+  privileged = false
+  role_map   = var.allowed_roles
```

nit: why not `allowed_role_map` for consistency with the module input ?

```suggestion
  role_map   = var.allowed_role_map
```

Other examples in other components

https://github.com/cloudposse/terraform-aws-components/blob/f51b6f31b196f1cfa3befe7f17024b1957bb8eeb/modules/eks/cluster/remote-state.tf#L7-L10

https://github.com/cloudposse/terraform-aws-components/blob/f51b6f31b196f1cfa3befe7f17024b1957bb8eeb/modules/ecr/main.tf#L1-L4

---

Ignored